api/api/views/viewsets/search.py

```diff
@@ -0,0 +1,64 @@
+# Search across all models
+
+from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from django.db.models import Q
+
+from api.models import (
+    Pentester,
+    Team,
+    Manager,
+    Pentester
+)
+from api.models.mission import Mission
+from api.models.vulns import Vulnerability
+
+
+class SearchView(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+
+    def search(self, request):
+        query = request.query_params.get('q', None)
+        if query is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        look_for_team = request.query_params.get('look_for_team', False)
+
+        # Search across all models
+        managers = Manager.objects.filter(
```

Comment:
Instead of having a list of `filter()` statements that repeat themselves, you can do something like this: 

```python
def fetch(method, searchable_fields, keywords):
    """will perform search query in a generic way"""

    if not keywords:
        return None

    words = keywords.split()
    query = Q()

    for fields in searchable_fields:
        for word in words:

            # no SQLi for you :%)
            if not word.isalnum():
                    continue

            # some input validation is necessary before executing this
            query.add(Q(**{f'{fields}__icontains': word}), Q.OR)
    
    # is_allowed_to_see() is not implemented in this example, 
    # but im sure you get the idea
    return [x for x in method().filter(query) if is_allowed_to_see(user, x)]
```

Then, you can call it like this: 
```python
user_keywords = ['first_name', 'last_name', 'email']
search = fetch(request.user, Manager.objects.all, user_keywords, request.GET.get('q'))
```

Reply:
then, it's pretty easy to add a `search_managers=True|False` parameter to check if the request should be ran or not, for even better optimizations B)